### PR TITLE
Add kernel parameters tests

### PR DESCRIPTION
**Pull request by aleksmesh**
Add tests according to [Test plan for parameter passing to kernels](https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/test_plans/kernels_parameters.asciidoc)

### CODE REVIEW

**Review by ProGTX — APPROVED** [review]

**Comment by fraggamuffin** [comment]
LGTM

**Review by keryell — APPROVED** [review]
Thanks.

File: tests/kernel/kernel_parameters.cpp [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+                  cgh);
+          named_kernel<decltype(acc_expected), T> kernel(acc_expected);
+          cgh.single_task(kernel);
+        })
+        .wait_and_throw();
+  }
+  sycl::queue queue;
+
+ public:
+  named_kernel_test() { auto queue = sycl_cts::util::get_cts_object::queue(); }
```
Comment (keryell) [inline tests/kernel/kernel_parameters.cpp:183]: I am confused about this line which appears many times. What is it for?

**Review by bader — APPROVED** [review]
LGTM, just a few minor style suggestions.

File: tests/common/value_operations.h [hunk 2]
```diff
@@ -50,8 +50,12 @@ using ArrayT = T[N];
  */
 template <typename T, typename U>
 void assign_value_or_even(T& left, const U& right) {
-  if constexpr (std::is_same_v<T, bool> ||
-                std::is_same_v<T, std::optional<bool>>)
+  if constexpr ((std::is_same_v<T, bool> ||
+                 std::is_same_v<
+                     T, std::optional<bool>>)&&(!std::is_same_v<U, bool> &&
```
[the remark this answers is not in the document]
Comment (aleksmesh) [reply tests/common/value_operations.h:55]: I'm sorry, but it is the result of clang-format-14

File: tests/kernel/kernel_parameters.cpp [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+//  limitations under the License.
+//
+//  Provides tests for kernel parameters
+//
+*******************************************************************************/
+#include "../common/common.h"
+#include "../common/type_coverage.h"
+
+namespace kernels_tests {
+constexpr size_t ARRAY_SIZE = 5;
```
[the remark this answers is not in the document]
Comment (aleksmesh) [reply tests/kernel/kernel_parameters.cpp]: Fixed.

File: tests/kernel/kernel_parameters.cpp (hunk 1 above)
Comment (aleksmesh) [reply tests/kernel/kernel_parameters.cpp]: Fixed.

File: tests/kernel/kernel_parameters.cpp [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+//  Provides tests for kernel parameters
+//
+*******************************************************************************/
+#include "../common/common.h"
+#include "../common/type_coverage.h"
+
+namespace kernels_tests {
```
[the remark this answers is not in the document]
Comment (aleksmesh) [reply tests/kernel/kernel_parameters.cpp]: Fixed.

**Review by keryell — APPROVED** [review]
Thanks.

**Pull request merged**